platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/ValueSanitizers.java

```diff
@@ -69,6 +84,8 @@ public void addAll(Iterable<?> values, ImmutableCollection.Builder<Object> dest)
     public static <T> ValueSanitizer<T> forType(Class<? extends T> targetType) {
         if (String.class.equals(targetType)) {
             return Cast.uncheckedCast(STRING_VALUE_SANITIZER);
+        } else if (Long.class.equals(targetType)) {
```

Comment:
💭   Should we also add the same to `collectorFor`?

Reply:
>Should we also add the same to collectorFor?

I think we should, fewer surprises for users.